File: Coding Ninjas/Ninja/Arithmetic Progression Queries.py
class NumArray:
	
	def __init__( self,arr ):
  
		self.arr = arr



	def update( self, l, r, val ):

		for i in range( r - l + 1 ):
			self.arr[ i + l - 1 ] += i + val



	def rangeSum( self, l, r ):

		return sum( self.arr[ l - 1 : r ] )

File: Coding Ninjas/Ninja/Arithmetic Progression Queries.java
public class NumArray
{

	int[] arr;



	// Initialize here
	NumArray
	( int[] arr )
	{
		this.arr = arr;
	}



	// Update operation
	public void update
	( int l, int r, int val )
	{
		for ( int i = l - 1; i < r; i++ )
		{
			arr[i] += val++;
		}
	}



	// Return the sum of subarray arr[l..r]
	public long rangeSum
	( int l, int r )
	{
		long ans = 0L;


		for ( int i = l - 1; i < r; i++ )
		{
			ans += arr[i];
		}


		return ans;
	}

}
